### PR TITLE
Remove global config from session

### PR DESCRIPTION
**Pull request by neubig**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

This PR removes global config from the session manager.

**Give a summary of what the PR does, explaining any non-trivial design decisions**

1. Removes global config from the session management code, and instead passes in variables
2. Converts some of the lower-level functions to take normal arguments instead of dictionaries
3. Initializes the session manager within the server listener

(I have run this and confirmed that it runs)

**Other references**
Part of #2758

### CODE REVIEW

**Review by iFurySt — APPROVED** [review]

**Comment by iFurySt** [comment]
LGTM

**Comment by neubig** [comment]
Mac unit tests are failing here, but they're also failing on main :(
https://github.com/OpenDevin/OpenDevin/actions/runs/9983889075/job/27592211025

**Comment by neubig** [comment]
Tests are failing due to https://github.com/OpenDevin/OpenDevin/issues/2992

File: opendevin/server/session/session.py [hunk 1]
```diff
@@ -67,8 +72,38 @@ async def _initialize_agent(self, data: dict):
...
+            ConfigType.LLM_MODEL, default_llm_config.model
+        )
+        default_llm_config.api_key = args.get(
+            ConfigType.LLM_API_KEY, default_llm_config.api_key
+        )
+
+        # TODO: override other LLM config & agent config groups (#2075)
+
+        llm = LLM(config=self.config.get_llm_config_from_agent(agent_cls))
+        agent = agent = Agent.get_cls(agent_cls)(llm)
```
Comment (tobitege) [inline opendevin/server/session/session.py:96]: Duplicate `agent =`

**Review by tobitege — CHANGES_REQUESTED** [review]
Nice job, but one typo.

**Review by tobitege — APPROVED** [review]

**Pull request merged**